src/EditorFeatures/Core.Cocoa/Microsoft.CodeAnalysis.EditorFeatures.Cocoa.csproj

```diff
@@ -17,6 +17,7 @@
     <Compile Include="..\Core.Wpf\SignatureHelp\**\*.cs" LinkBase="SignatureHelp" />
     <Compile Include="..\Core.Wpf\Suggestions\**\*.cs" LinkBase="Suggestions" />
     <Compile Include="..\Core.Wpf\IDebuggerTextView2.cs" Link="IDebuggerTextView2.cs" />
+    <Compile Include="..\Core.Wpf\Tags\IImageMonikerService.cs" LinkBase="Tags" />
```

Comment:
I don't love the idea of adding a single interface to Cocoa.dll which will never be implemented, but this avoids peppering `#if !COCOA` around the place so its a bit neater. Happy to change to that model if people prefer though.

Reply:
I think this is fine

Reply:
So why is this being added like this? I would have thought the whole point of this was to get rid of this on the Cocoa side of things.

Reply:
This is just to avoid putting various `#if COCOA` blocks in here: https://github.com/dotnet/roslyn/blob/main/src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSourceProvider.cs#L64

I'm happy to do that if its preferred, I don't feel strongly either way. This still means that someone has to try extra hard to use it on the cocoa side, rather than accidentally doing it because they reference EditorFeatures, but certainly the `#if`s are a stronger statement.

Reply:
Oh, I hadn't realized that this way we'll still have ImageMoniker in EF.Cocoa.dll. Maybe it's indeed better to do `#if COCOA` on the Wpf side? If we can remove the Imaging reference entirely from Cocoa.dll that would be an extra safeguard to avoid accidentally using it.

Reply:
Hmm sure would be nice to get rid of the interop reference here:

![image](https://user-images.githubusercontent.com/679326/114459260-2352e680-9b95-11eb-8e98-0827685af063.png)

Reply:
So it's not just EditorFeatures.Cocoa having the reference, the editor APIs themselves still do: even on the cocoa side, ISuggestedAction.ImageMoniker exists. The EditorFeatures layer _cannot_ get rid of ImageMoniker references on it's own -- we need the Editor to fix layering first.

Reply:
Yeah, sadly we can't entirely get rid of ImageMoniker, but at least with this change we don't have it as an embedded type in EditorFeatures.dll itself.

Reply:
As long as @KirillOsenkov is OK with this.

Reply:
Ah, yes, ISuggestedAction. I knew we can't have nice things! And here I was hopeful for a brief moment. OK yeah if we can't get rid of the reference anyway than I'm OK with whatever, sorry for the confusion.

Reply:
Will we have the embedded type in EF.Cocoa.dll?

Reply:
No, that was already turned off in https://github.com/dotnet/roslyn/pull/51872